src/server/v2.0/handler/project.go

```diff
@@ -99,11 +99,17 @@ func (a *projectAPI) CreateProject(ctx context.Context, params operation.CreateP
 	}
 
 	secCtx, _ := security.FromContext(ctx)
-	if onlyAdmin && !(a.isSysAdmin(ctx, rbac.ActionCreate) || secCtx.IsSolutionUser()) {
+	if onlyAdmin && secCtx.Name() != "robot" && !(a.isSysAdmin(ctx, rbac.ActionCreate) || secCtx.IsSolutionUser()) {
 		log.Errorf("Only sys admin can create project")
 		return a.SendError(ctx, errors.ForbiddenError(nil).WithMessage("Only system admin can create project"))
 	}
 
+	// always check if robots have permissions to create projects
```

Comment:
if !onlyAdmin {
    if it's a project level robot
        return errors.ForbiddenError
}

Reply:
is there a method to check, if the robot acocunt scope is system or project level? i didn'd saw something like a getLevel method.

Reply:
@heww is it better to use `Can` to define whether it's a system level robot?

Reply:
Or here we can leverage the robot name to query robot, there is an attribute to indicate robot scope.

Reply:
the robot context have a isSystemLevel property, but it was not exported. i exported the property with a getter and used that getter.

Reply:
Another trick way, as the robot name has a pattern, the project level name should be {projectname}+robotname, which is difference with system level robot. And `+` is illegal character when to issue a new robot, so here we can use whether contains `+` to judge.

Reply:
imho using a character to identify if a robot is system or project level is not a good idea. if you set the prefix for example to "robot+" the check will fail. the robot security context already has a property which identifies if a robot is system or project level so why not use it?